extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/stream/YoutubeStreamExtractorDefaultTest.java

```diff
@@ -249,15 +250,18 @@ public void testGetDescription() throws ParsingException {
 
         @Test
         public void testGetFullLinksInDescription() throws ParsingException {
-            assertTrue(extractor.getDescription().getContent().contains("https://www.youtube.com/watch?v=X7FLCHVXpsA&amp;list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34"));
-            assertTrue(extractor.getDescription().getContent().contains("https://www.youtube.com/watch?v=Lqv6G0pDNnw&amp;list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34"));
-            assertTrue(extractor.getDescription().getContent().contains("https://www.youtube.com/watch?v=XxaRBPyrnBU&amp;list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34"));
-            assertTrue(extractor.getDescription().getContent().contains("https://www.youtube.com/watch?v=U-9tUEOFKNU&amp;list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34"));
-
-            assertFalse(extractor.getDescription().getContent().contains("https://youtu.be/X7FLCHVXpsA?list=PL7..."));
-            assertFalse(extractor.getDescription().getContent().contains("https://youtu.be/Lqv6G0pDNnw?list=PL7..."));
-            assertFalse(extractor.getDescription().getContent().contains("https://youtu.be/XxaRBPyrnBU?list=PL7..."));
-            assertFalse(extractor.getDescription().getContent().contains("https://youtu.be/U-9tUEOFKNU?list=PL7..."));
+            Description description = extractor.getDescription();
+            String content = description.getContent();
+            if (description.getType() == Description.HTML) {
+                assertTrue(content.contains("https://www.youtube.com/watch?v=X7FLCHVXpsA&list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34"));
+                assertTrue(content.contains("https://www.youtube.com/watch?v=Lqv6G0pDNnw&amp;list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34"));
+                assertTrue(content.contains("https://www.youtube.com/watch?v=XxaRBPyrnBU&amp;list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34"));
+                assertTrue(content.contains("https://www.youtube.com/watch?v=U-9tUEOFKNU&amp;list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34"));
+            }
```

Comment:
If I understood correctly, the HTML description has the same shortened links as the JSON microformat description. So  this should be (note there is no `if`, i.e. no need to distinguish between formats):
```
                assertTrue(content.contains("https://youtu.be/X7FLCHVXpsA?list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34"));
                assertTrue(content.contains("https://youtu.be/Lqv6G0pDNnw?list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34"));
                assertTrue(content.contains("https://youtu.be/XxaRBPyrnBU?list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34"));
                assertTrue(content.contains("https://youtu.be/U-9tUEOFKNU?list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34"));
```

Reply:
test this:
```java
@Test
public void test() throws IOException, ReCaptchaException {
    Downloader dl = NewPipe.getDownloader();
    Response response = dl.get("https://www.youtube.com/watch?v=cV5TjZCJkuA");
    Document doc = Jsoup.parse(response.responseBody());
    System.out.println(doc.select("#eow-description"));
}
```
Basically, all links are like that
```<a href="/watch?v=X7FLCHVXpsA&amp;list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34" class="yt-uix-sessionlink       spf-link " data-sessionlink="itct=CCwQ6TgYACITCNb42KP44ucCFZYWBgAdKaEKRA">https://youtu.be/X7FLCHVXpsA?list=PL7...</a>```
If we don't use ``parseHtmlAndGetFullLinks``, we get this: <a href="/watch?v=X7FLCHVXpsA&amp;list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34" class="yt-uix-sessionlink       spf-link " data-sessionlink="itct=CCwQ6TgYACITCNb42KP44ucCFZYWBgAdKaEKRA">https://youtu.be/X7FLCHVXpsA?list=PL7...</a> broken because the link in href needs the domain, which is not youtube because we aren't youtube.

That's why we use ``parseHtmlAndGetFullLinks``, to gather what is in href, and fix the link.

For JSON, it's not needed, because it gives the proper link directly.

If I do your requested changes, it won't work because HTML gives full links, JSON shortened.
What we don't want, is to have broken shortened links, like this <a href="/watch?v=X7FLCHVXpsA&amp;list=PL7u4lWXQ3wfI_7PgX0C-VTiwLeu0S4v34" class="yt-uix-sessionlink       spf-link " data-sessionlink="itct=CCwQ6TgYACITCNb42KP44ucCFZYWBgAdKaEKRA">https://youtu.be/X7FLCHVXpsA?list=PL7...</a> and test if the links are good.

So we need a if.

Reply:
Can't you just do `new URL(new URL(whateverFunctionGetsTheUrlOfTheVideo()), whateverVariableHasTheHref)`? Or is that exactly what that function does?

Reply:
Which function? The test one? If yes, you are right